buildpack.md

```diff
@@ -1138,6 +1144,39 @@ Each `key`:
 ## Deprecations
 This section describes all the features that are deprecated.
 
+### buildpack.toml (TOML) `stacks` Array
+
+_Deprecated in Buildpack API 0.10._
+
+The `stacks` array is deprecated.
+
+```toml
+[[stacks]]
+id = "<stack ID>"
+mixins = ["<mixin name>"]
+```
+
+Each stack in `stacks` either:
+- MUST identify a compatible stack:
+    - `id` MUST be set to a [valid stack ID](https://github.com/buildpacks/spec/blob/main/platform.md#stack-id).
+    - `mixins` MAY contain one or more mixin names.
+- Or MUST indicate compatibility with any stack:
+    - `id` MUST be set to the special value `"*"`.
+    - `mixins` MUST be empty.
+
+If an `order` is specified, then `stacks` MUST NOT be specified.
+
+Tools reading `buildpack.toml` will translate any section that sets `stacks.id = "io.buildpacks.stacks.bionic"` to:
```

Comment:
Could we allow builders to provide a stack<>target mapping? Or maybe allow the platform to specify this for b/w compat?

Reply:
@samj1912 looking at the [table](https://github.com/buildpacks/spec/pull/336#issuecomment-1454099934) below, are you thinking of the case where a new platform might want to do some sort of validation around what target distributions an old buildpack might support, but only has the stack ID available? 

If so, I imagine we could add that mapping by allowing [stack] in builder.toml to contain (optional) targets:

```
# Stack that will be used by the builder
[stack]
id = "io.buildpacks.samples.stacks.bionic"
run-image = "cnbs/sample-stack-run:bionic"
build-image = "cnbs/sample-stack-build:bionic"
[[targets.distributions]]   <--- Targets this stack ID will satisfy if we find the stack ID in buildpack.toml; does it make sense for a single stack to satisfy multiple targets?
name = "<OS distribution name>"
version = "version"
```

Reply:
I could be wrong but I think the ability to derive the os/arch from the ./bin/build extension (or lack thereof) might get us most of the way there, and distributions is the only gap left to patch. My guess is that stack IDs in the wild don't contain data as detailed as "arch variant".

Reply:
Worth noting, we're planning to do target validation at the point of detection in the lifecycle, but we're treating missing optional target information as "wildcard" (cc @joe-kimmel-vmw). The above approach would enable platforms to overlay stricter validation if they desire, but it would be optional and I don't think it makes sense to repeat in the lifecycle.

It's also possible that I completely misunderstood what you meant...

Reply:
Discussed in 3/9/23 Working Group, along with this table to summarize the backwards compat strategy: https://docs.google.com/spreadsheets/d/1_lUDh8V_A6SHV9w2ioGwqAu821WWaItSYiUKqrlmr_w/edit#gid=240821534

We think that reading values from /etc/os-release will fill the last remaining gaps.